peer-test-suite/python/logic.py

```
@@ -1,3 +1,16 @@
-    # missing None checks; assume user has 'age'
-    return 2025 - user['birth_year']
+
+def analyze_data(data, cache=[]):
+    # High Level Error: Mutable default argument 'cache' retains state between calls
+    # leading to unexpected data leakage across different function calls.
+    cache.append(data)
+    return len(cache)
+
+def main():
+    # Simple Error: Syntax error (Case sensitivity in function name)
+    Print("Starting analysis")
```

Ruff (0.14.10)

Undefined name `Print`

(F821)